// @shopify/polaris/build/cjs/components/Modal/components/Section/Section.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var css = require('../../../../utilities/css.js');
var Section$1 = require('./Section.scss.js');
var Box = require('../../../Box/Box.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

function Section({
  children,
  flush = false,
  subdued = false,
  titleHidden = false
}) {
  const className = css.classNames(Section$1["default"].Section, titleHidden && Section$1["default"].titleHidden);
  return /*#__PURE__*/React__default["default"].createElement("div", {
    className: className
  }, /*#__PURE__*/React__default["default"].createElement(Box.Box, Object.assign({
    as: "section",
    padding: flush ? '0' : '5'
  }, titleHidden && {
    paddingInlineEnd: '0'
  }, subdued && {
    background: 'bg-subdued'
  }), children));
}

exports.Section = Section;
